playwright/_impl/_stream.py

```diff
@@ -26,9 +27,8 @@ def __init__(
         super().__init__(parent, type, guid, initializer)
 
     async def save_as(self, path: Union[str, Path]) -> None:
-        with open(path, mode="wb") as file:
-            while True:
-                binary = await self._channel.send("read")
-                if not binary:
-                    break
-                file.write(base64.b64decode(binary))
+        while True:
+            binary = await self._channel.send("read")
+            if not binary:
+                break
+            await async_writefile(path, "wb", base64.b64decode(binary))
```

Comment:
This looks like a bug, you re-write this file with the last chunk over and over.